homeassistant/components/fully_kiosk/switch.py

```diff
@@ -0,0 +1,119 @@
+"""Fully Kiosk Browser switch."""
+from __future__ import annotations
+
+from collections.abc import Callable
+from dataclasses import dataclass
+from typing import Any
+
+from fullykiosk import FullyKiosk
+
+from homeassistant.components.switch import SwitchEntity, SwitchEntityDescription
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import EntityCategory
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+from .coordinator import FullyKioskDataUpdateCoordinator
+from .entity import FullyKioskEntity
+
+
+@dataclass
+class FullySwitchEntityDescriptionMixin:
+    """Fully Kiosk Browser switch entity description mixin."""
+
+    on_action: Callable[[FullyKiosk], Any]
+    off_action: Callable[[FullyKiosk], Any]
+    is_on_fn: Callable[[FullyKioskDataUpdateCoordinator], Any]
```

Comment:
We don't have to pass the coordinator, only its data.

```suggestion
    is_on_fn: Callable[[dict[str, Any], Any]
```